### PR TITLE
fix(hass): dimmers and rgb dimmers

### LINKED ISSUE #280: [question] GE fan switches
This is the Hass device JSON that was created by zjs2m for the GE fan I have, which doesn't work with HA.
```
{
  "type": "fan",
  "object_id": "dimmer",
  "values": [
    "38-0-currentValue",
    "38-0-targetValue"
  ],
  "discovery_payload": {
    "command_topic": "zwave2jsmqtt/master_bedroom_fan/38/0/currentValue/set",
    "speed_command_topic": "zwave2jsmqtt/master_bedroom_fan/38/0/targetValue/set",
    "speed_state_topic": "zwave2jsmqtt/master_bedroom_fan/38/0/currentValue",
    "state_topic": "zwave2jsmqtt/master_bedroom_fan/38/0/currentValue",
    "speeds": [
      "off",
      "low",
      "medium",
      "high"
    ],
    "payload_low_speed": 24,
    "payload_medium_speed": 50,
    "payload_high_speed": 99,
    "payload_off": 0,
    "payload_on": 255,
    "state_value_template": "{% if (value_json.value | int) == 0 %} 0 {% else %} 255 {% endif %}",
    "speed_value_template": "{% if (value_json.value | int) == 0 %} 0 {% elif (value_json.value | int) <= 32 %} 24 {% elif (value_json.value | int) <= 66 %} 50 {% elif (value_json.value | int) <= 99 %} 99 {% endif %}",
    "device": {
      "identifiers": [
        "zwavejs2mqtt_0x_node17"
      ],
      "manufacturer": "GE/Jasco",
      "model": "In-Wall Smart Fan Control (ZW4002)",
      "name": "master_bedroom_fan",
      "sw_version": "5.22"
    },
    "name": "master_bedroom_fan_dimmer",
    "unique_id": "zwavejs2mqtt_0x_Node17_dimmer"
  },
  "discoveryTopic": "fan/master_bedroom_fan/dimmer/config",
  "persistent": false,
  "ignoreDiscovery": false,
  "id": "fan_dimmer"
}
```

After changing the "command_topic" to "targetValue/set" seems to work for turning it on and off in HA and showing "99" as a Current value in the Multilevel switch info for the node (in zjs2m) but changing the speed shows 19 for low and 38 for medium instead of the 24 and 50 like the in the JSON.

So I'm just wondering if I need to change anything else to make this fan switch work properly.

### LINKED ISSUE #285: Dimmers no longer go to 100% in rc0
**Version**

Build/Run method
- [ ] Docker
- [ ] PKG
- [X] Manually built (git clone - npm install - npm run build ) 

zwavejs2mqtt version: 1.0.0-rc.0

**Describe the bug**
Attempting to set the brightness of a standard (non-RGB) dimmer no longer will go to 100%.

**To Reproduce**
1. Deploy zwavejs2mqtt rc.0
2. Set the brightness of a dimmer to 87% in HASS
3. Observe that the light is much dimmer than expected
4. Observe that the dim level displayed in HA has snapped down to a much lower level than selected
5. In the zwavejs2mqtt log:
2021-01-20 03:18:46.390 INFO MQTT: Message received on homeassistant/DiningRoom/13/38/0/targetValue/set, '87'
2021-01-20 03:18:46.401 INFO ZWAVE: Writing 34 to 38-0-targetValue

**Expected behavior**
The dimmer should dim to the value selected. Swapping out zwavejs2mqtt and keeping everything else the same (HA version, dimmer in question, MQTT settings, zwave-js version) yields a dim of 87%

**Additional context**
[node_13.zip](https://github.com/zwave-js/zwavejs2mqtt/files/5842668/node_13.zip)

### LINKED ISSUE #286: With latest RGB updates, existing RGB dimmer stops working
**Version**

Build/Run method
- [ ] Docker
- [ ] PKG
- [X] Manually built (git clone - npm install - npm run build ) 

zwavejs2mqtt version: 1.0.0-rc.0

**Describe the bug**
The "master dimmer" of an existing RGBW dimmer (Zooz zen31) stops working with the latest update, the entity must be removed and re-added to HA

**To Reproduce**
1. Get set up with the 1.0.0-beta zwavejs2mqtt
2. Ensure an RGBW dimmer has been discovered. It will appear in HA as 5 dimmers: master=1, red=2, green=3, blue=4, white=5
3. Update to 1.0.0-rc.0
4. Attempt to control the dimmer. Master dimmer control will have stopped working completely

**Expected behavior**
I would expect that the dimmer will continue to function at worst, and will be updated to support RGB controls at best

**Additional context**
From the HA log:
2021-01-20 01:06:10 ERROR (MainThread) [homeassistant.components.light] Platform mqtt does not generate unique IDs. ID zwavejs2mqtt_0xcdbd925c_25-38-1-currentValue already exists - ignoring light.familyroom_tvlight_rgb_dimmer

[node_25_beta.zip](https://github.com/zwave-js/zwavejs2mqtt/files/5842675/node_25_beta.zip)
[node_25_rc.zip](https://github.com/zwave-js/zwavejs2mqtt/files/5842676/node_25_rc.zip)

### LINKED ISSUE #287: RGBW Dimmer: Can't control white
**Version**

Build/Run method
- [ ] Docker
- [ ] PKG
- [X] Manually built (git clone - npm install - npm run build ) 

zwavejs2mqtt version: 1.0.0-rc.0

**Describe the bug**
White does not show up on the color picker for an RGBW dimmer (Zooz zen31)

**To Reproduce**
1. Get set up with the 1.0.0-rc.0 zwavejs2mqtt
2. Rediscover a zooz zen31 RGBW dimmer
3. Open the RGB dimmer control
4. Observe that only the color dial is present, the white control is missing

**Expected behavior**
I would expect that there would be a white control on the display

**Additional context**
[node_25_rc.zip](https://github.com/zwave-js/zwavejs2mqtt/files/5842676/node_25_rc.zip)
How this dimmer shows up:
image

How I would expect it to appear (this is a tasmota RGBW wifi light, not my image)
image

### LINKED ISSUE #288: RGBW Dimmer: On/Off control does not function
**Version**

Build/Run method
- [ ] Docker
- [ ] PKG
- [X] Manually built (git clone - npm install - npm run build ) 

zwavejs2mqtt version: 1.0.0-rc.0

**Describe the bug**
On/Off control fails to function for an RGBW dimmer (Zooz zen31)

**To Reproduce**
1. Get set up with the 1.0.0-rc.0 zwavejs2mqtt
2. Rediscover a zooz zen31 RGBW dimmer
3. Open the RGB dimmer control
4. Attempt to turn on or off the light with the on/off control
5. Observe that nothing has happened

**Expected behavior**
I would expect that the light would turn on and off

**Additional context**
From zwavejs2mqtt log:
2021-01-20 13:30:29.818 INFO MQTT: Message received on homeassistant/FamilyRoom/25/38/1/targetValue/set, 'ON'
2021-01-20 13:30:29.827 INFO ZWAVE: Writing 'ON' to 38-1-targetValue
2021-01-20 13:30:29.833 ERROR ZWAVE: Error while writing 'ON' on 38-1-targetValue: Command not valid for Multilevel Switch
2021-01-20 13:30:29.836 ERROR ZWAVE: Unable to write 'ON' on 38-1-targetValue

[node_25_rc.zip](https://github.com/zwave-js/zwavejs2mqtt/files/5842676/node_25_rc.zip)

### PR DESCRIPTION
**Pull request by robertsLando**
Fixes #280 
Fixes #285 
Fixes #288
Fixes #287 
Fixes #286

### CODE REVIEW

File: lib/Gateway.js [hunk 1]
```diff
@@ -647,6 +647,49 @@ function setDiscoveryValue (payload, prop, node) {
...
+      cfg.discovery_payload.rgb_state_topic + '/set'
+    cfg.values = [rgbID]
+
+    const whiteID = `51-${valueId.endpoint || 0}-currentColor-0`
+    const whiteValue = node.values[whiteID]
+
+    if (whiteValue) {
+      const whiteIDWrite = `51-${valueId.endpoint || 0}-targetColor-0`
+      cfg.discovery_payload.white_value_state_topic = this.mqtt.getTopic(
+        this.valueTopic(node, whiteID)
```
Comment (robertsLando) [inline lib/Gateway.js:676]:
```suggestion
        this.valueTopic(node, whiteValue)
```

File: lib/Gateway.js [hunk 2]
```diff
@@ -647,6 +647,49 @@ function setDiscoveryValue (payload, prop, node) {
...
+    )
+    cfg.discovery_payload.rgb_command_topic =
+      cfg.discovery_payload.rgb_state_topic + '/set'
+    cfg.values = [rgbID]
+
+    const whiteID = `51-${valueId.endpoint || 0}-currentColor-0`
+    const whiteValue = node.values[whiteID]
+
+    if (whiteValue) {
+      const whiteIDWrite = `51-${valueId.endpoint || 0}-targetColor-0`
```
Comment (robertsLando) [inline lib/Gateway.js:674]:
```suggestion
      const whiteIDWrite = `51-${valueId.endpoint || 0}-targetColor-0`
      const whiteValueWrite = node.values[whiteIDWrite]
```

File: lib/Gateway.js [hunk 3]
```diff
@@ -647,6 +647,49 @@ function setDiscoveryValue (payload, prop, node) {
...
+      cfg.discovery_payload.white_value_state_topic = this.mqtt.getTopic(
+        this.valueTopic(node, whiteID)
+      )
+      cfg.discovery_payload.white_value_scale = whiteValue.max
+      cfg.discovery_payload.white_value_command_topic = this.mqtt.getTopic(
+        this.valueTopic(node, whiteIDWrite),
```
Comment (robertsLando) [inline lib/Gateway.js:680]:
```suggestion
        this.valueTopic(node, whiteValueWrite),
```

**Comment by towerhand** [comment]
Is there an easy way to pull this fix with docker compose? Or would it be easier to wait until is merged and pull it with the dev tag?

**Comment by jcam** [comment]
This PR fixes the last of my issues. You may want to further update it so the 'scale' is set programmatically
https://github.com/zwave-js/zwavejs2mqtt/pull/295

**Comment by blhoward2** [comment]
This also works fine for me, though I wasn't yet on master so I can't actually confirm that I had the problem...

**Comment by robertsLando** [comment]
@jcam Is there something else missing? Are all the problems fixed now?

**Pull request merged**